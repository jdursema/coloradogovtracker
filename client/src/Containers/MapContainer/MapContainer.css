.main-container {
  max-width: 1200px;
}

.map-container-content {
  display:flex;
  flex-direction: row;
}

.map-container h1 {
  text-align: center;
}

.map-container h3 {
  text-align: center;
  font-weight: 300;
}

.fund-head {
  border-bottom: 1px solid gray;
}

.map {
  width: 70%;
}

.map-totals {
  width: 30%;
  display:flex;
  align-items: center;
  flex-direction: column;
  margin-top: 60px;
}

ul.top-states {
  list-style-type: none;
}

@media screen and (max-width: 730px) {
    .map-container-content {
      flex-direction: column;
    }

  .map {
    width: 100%;
  }

  .map-totals {
    width: 100%;
  }
}