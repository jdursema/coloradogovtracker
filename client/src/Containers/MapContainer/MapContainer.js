import React, { Component } from 'react';
import DataMap from '../Map/Map'
import './MapContainer.css'


const MapContainer = () => {

 
    return (
      <div className = "map-container" id="container-map">
        <h1> Follow the money </h1>
        <h3> Contributions to the 2018 Colorado governor's race have come from every state </h3>
        <div className = "map-container-content">
          <div className = "map">
            <DataMap />
          </div>
          <div className = "map-totals">
            <h3 className = "fund-head"> Top Fundraising States </h3>
            <ul className = "top-states">
              <li> <span className="light">Colorado:</span> $5,579,829 </li>
              <li> <span className="light">California:</span> $345,014 </li>
              <li> <span className="light">New York:</span> $317,925</li>
              <li> <span className="light">Connecticut:</span> $142,720</li>
               <li> <span className="light">Massachussets:</span> $105,922</li>
            </ul>
            <span className = "small"> *Hover over map to see all totals </span>
          </div>
        </div>
         
      </div>
    )
}

export default MapContainer;