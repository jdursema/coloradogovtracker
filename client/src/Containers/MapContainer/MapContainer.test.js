import MapContainer from './MapContainer'
import React from 'react';
import {shallow} from 'enzyme';

describe('Map container', () => {
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = shallow(
      <MapContainer />
    )
  })

  it('should render', () => {
    expect(renderedComponent).toBeDefined()
  })

  it('should match the snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  })

  it('should have a class map container and map totals', () => {
    expect(renderedComponent.find('.map-container').length).toEqual(1);
    expect(renderedComponent.find('.map-totals').length).toEqual(1);
  })

})