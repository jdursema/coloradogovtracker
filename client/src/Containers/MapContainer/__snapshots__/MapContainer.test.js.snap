// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map container should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           Follow the money 
</h1>,
        <h3>
           Contributions to the 2018 Colorado governor's race have come from every state 
</h3>,
        <div
          className="map-container-content"
>
          <div
                    className="map"
          >
                    <DataMap />
          </div>
          <div
                    className="map-totals"
          >
                    <h3
                              className="fund-head"
                    >
                               Top Fundraising States 
                    </h3>
                    <ul
                              className="top-states"
                    >
                              <li>
                                         
                                        <span
                                                  className="light"
                                        >
                                                  Colorado:
                                        </span>
                                         $5,579,829 
                              </li>
                              <li>
                                         
                                        <span
                                                  className="light"
                                        >
                                                  California:
                                        </span>
                                         $345,014 
                              </li>
                              <li>
                                         
                                        <span
                                                  className="light"
                                        >
                                                  New York:
                                        </span>
                                         $317,925
                              </li>
                              <li>
                                         
                                        <span
                                                  className="light"
                                        >
                                                  Connecticut:
                                        </span>
                                         $142,720
                              </li>
                              <li>
                                         
                                        <span
                                                  className="light"
                                        >
                                                  Massachussets:
                                        </span>
                                         $105,922
                              </li>
                    </ul>
                    <span
                              className="small"
                    >
                               *Hover over map to see all totals 
                    </span>
          </div>
</div>,
      ],
      "className": "map-container",
      "id": "container-map",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Follow the money ",
        },
        "ref": null,
        "rendered": " Follow the money ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Contributions to the 2018 Colorado governor's race have come from every state ",
        },
        "ref": null,
        "rendered": " Contributions to the 2018 Colorado governor's race have come from every state ",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="map"
>
              <DataMap />
</div>,
            <div
              className="map-totals"
>
              <h3
                            className="fund-head"
              >
                             Top Fundraising States 
              </h3>
              <ul
                            className="top-states"
              >
                            <li>
                                           
                                          <span
                                                        className="light"
                                          >
                                                        Colorado:
                                          </span>
                                           $5,579,829 
                            </li>
                            <li>
                                           
                                          <span
                                                        className="light"
                                          >
                                                        California:
                                          </span>
                                           $345,014 
                            </li>
                            <li>
                                           
                                          <span
                                                        className="light"
                                          >
                                                        New York:
                                          </span>
                                           $317,925
                            </li>
                            <li>
                                           
                                          <span
                                                        className="light"
                                          >
                                                        Connecticut:
                                          </span>
                                           $142,720
                            </li>
                            <li>
                                           
                                          <span
                                                        className="light"
                                          >
                                                        Massachussets:
                                          </span>
                                           $105,922
                            </li>
              </ul>
              <span
                            className="small"
              >
                             *Hover over map to see all totals 
              </span>
</div>,
          ],
          "className": "map-container-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <DataMap />,
              "className": "map",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="fund-head"
>
                   Top Fundraising States 
</h3>,
                <ul
                  className="top-states"
>
                  <li>
                                     
                                    <span
                                                      className="light"
                                    >
                                                      Colorado:
                                    </span>
                                     $5,579,829 
                  </li>
                  <li>
                                     
                                    <span
                                                      className="light"
                                    >
                                                      California:
                                    </span>
                                     $345,014 
                  </li>
                  <li>
                                     
                                    <span
                                                      className="light"
                                    >
                                                      New York:
                                    </span>
                                     $317,925
                  </li>
                  <li>
                                     
                                    <span
                                                      className="light"
                                    >
                                                      Connecticut:
                                    </span>
                                     $142,720
                  </li>
                  <li>
                                     
                                    <span
                                                      className="light"
                                    >
                                                      Massachussets:
                                    </span>
                                     $105,922
                  </li>
</ul>,
                <span
                  className="small"
>
                   *Hover over map to see all totals 
</span>,
              ],
              "className": "map-totals",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Top Fundraising States ",
                  "className": "fund-head",
                },
                "ref": null,
                "rendered": " Top Fundraising States ",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li>
                       
                      <span
                                            className="light"
                      >
                                            Colorado:
                      </span>
                       $5,579,829 
</li>,
                    <li>
                       
                      <span
                                            className="light"
                      >
                                            California:
                      </span>
                       $345,014 
</li>,
                    <li>
                       
                      <span
                                            className="light"
                      >
                                            New York:
                      </span>
                       $317,925
</li>,
                    <li>
                       
                      <span
                                            className="light"
                      >
                                            Connecticut:
                      </span>
                       $142,720
</li>,
                    <li>
                       
                      <span
                                            className="light"
                      >
                                            Massachussets:
                      </span>
                       $105,922
</li>,
                  ],
                  "className": "top-states",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        <span
                          className="light"
>
                          Colorado:
</span>,
                        " $5,579,829 ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Colorado:",
                          "className": "light",
                        },
                        "ref": null,
                        "rendered": "Colorado:",
                        "type": "span",
                      },
                      " $5,579,829 ",
                    ],
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        <span
                          className="light"
>
                          California:
</span>,
                        " $345,014 ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "California:",
                          "className": "light",
                        },
                        "ref": null,
                        "rendered": "California:",
                        "type": "span",
                      },
                      " $345,014 ",
                    ],
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        <span
                          className="light"
>
                          New York:
</span>,
                        " $317,925",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "New York:",
                          "className": "light",
                        },
                        "ref": null,
                        "rendered": "New York:",
                        "type": "span",
                      },
                      " $317,925",
                    ],
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        <span
                          className="light"
>
                          Connecticut:
</span>,
                        " $142,720",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Connecticut:",
                          "className": "light",
                        },
                        "ref": null,
                        "rendered": "Connecticut:",
                        "type": "span",
                      },
                      " $142,720",
                    ],
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        <span
                          className="light"
>
                          Massachussets:
</span>,
                        " $105,922",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Massachussets:",
                          "className": "light",
                        },
                        "ref": null,
                        "rendered": "Massachussets:",
                        "type": "span",
                      },
                      " $105,922",
                    ],
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " *Hover over map to see all totals ",
                  "className": "small",
                },
                "ref": null,
                "rendered": " *Hover over map to see all totals ",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             Follow the money 
</h1>,
          <h3>
             Contributions to the 2018 Colorado governor's race have come from every state 
</h3>,
          <div
            className="map-container-content"
>
            <div
                        className="map"
            >
                        <DataMap />
            </div>
            <div
                        className="map-totals"
            >
                        <h3
                                    className="fund-head"
                        >
                                     Top Fundraising States 
                        </h3>
                        <ul
                                    className="top-states"
                        >
                                    <li>
                                                 
                                                <span
                                                            className="light"
                                                >
                                                            Colorado:
                                                </span>
                                                 $5,579,829 
                                    </li>
                                    <li>
                                                 
                                                <span
                                                            className="light"
                                                >
                                                            California:
                                                </span>
                                                 $345,014 
                                    </li>
                                    <li>
                                                 
                                                <span
                                                            className="light"
                                                >
                                                            New York:
                                                </span>
                                                 $317,925
                                    </li>
                                    <li>
                                                 
                                                <span
                                                            className="light"
                                                >
                                                            Connecticut:
                                                </span>
                                                 $142,720
                                    </li>
                                    <li>
                                                 
                                                <span
                                                            className="light"
                                                >
                                                            Massachussets:
                                                </span>
                                                 $105,922
                                    </li>
                        </ul>
                        <span
                                    className="small"
                        >
                                     *Hover over map to see all totals 
                        </span>
            </div>
</div>,
        ],
        "className": "map-container",
        "id": "container-map",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Follow the money ",
          },
          "ref": null,
          "rendered": " Follow the money ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Contributions to the 2018 Colorado governor's race have come from every state ",
          },
          "ref": null,
          "rendered": " Contributions to the 2018 Colorado governor's race have come from every state ",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="map"
>
                <DataMap />
</div>,
              <div
                className="map-totals"
>
                <h3
                                className="fund-head"
                >
                                 Top Fundraising States 
                </h3>
                <ul
                                className="top-states"
                >
                                <li>
                                                 
                                                <span
                                                                className="light"
                                                >
                                                                Colorado:
                                                </span>
                                                 $5,579,829 
                                </li>
                                <li>
                                                 
                                                <span
                                                                className="light"
                                                >
                                                                California:
                                                </span>
                                                 $345,014 
                                </li>
                                <li>
                                                 
                                                <span
                                                                className="light"
                                                >
                                                                New York:
                                                </span>
                                                 $317,925
                                </li>
                                <li>
                                                 
                                                <span
                                                                className="light"
                                                >
                                                                Connecticut:
                                                </span>
                                                 $142,720
                                </li>
                                <li>
                                                 
                                                <span
                                                                className="light"
                                                >
                                                                Massachussets:
                                                </span>
                                                 $105,922
                                </li>
                </ul>
                <span
                                className="small"
                >
                                 *Hover over map to see all totals 
                </span>
</div>,
            ],
            "className": "map-container-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <DataMap />,
                "className": "map",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="fund-head"
>
                     Top Fundraising States 
</h3>,
                  <ul
                    className="top-states"
>
                    <li>
                                         
                                        <span
                                                            className="light"
                                        >
                                                            Colorado:
                                        </span>
                                         $5,579,829 
                    </li>
                    <li>
                                         
                                        <span
                                                            className="light"
                                        >
                                                            California:
                                        </span>
                                         $345,014 
                    </li>
                    <li>
                                         
                                        <span
                                                            className="light"
                                        >
                                                            New York:
                                        </span>
                                         $317,925
                    </li>
                    <li>
                                         
                                        <span
                                                            className="light"
                                        >
                                                            Connecticut:
                                        </span>
                                         $142,720
                    </li>
                    <li>
                                         
                                        <span
                                                            className="light"
                                        >
                                                            Massachussets:
                                        </span>
                                         $105,922
                    </li>
</ul>,
                  <span
                    className="small"
>
                     *Hover over map to see all totals 
</span>,
                ],
                "className": "map-totals",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Top Fundraising States ",
                    "className": "fund-head",
                  },
                  "ref": null,
                  "rendered": " Top Fundraising States ",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li>
                         
                        <span
                                                className="light"
                        >
                                                Colorado:
                        </span>
                         $5,579,829 
</li>,
                      <li>
                         
                        <span
                                                className="light"
                        >
                                                California:
                        </span>
                         $345,014 
</li>,
                      <li>
                         
                        <span
                                                className="light"
                        >
                                                New York:
                        </span>
                         $317,925
</li>,
                      <li>
                         
                        <span
                                                className="light"
                        >
                                                Connecticut:
                        </span>
                         $142,720
</li>,
                      <li>
                         
                        <span
                                                className="light"
                        >
                                                Massachussets:
                        </span>
                         $105,922
</li>,
                    ],
                    "className": "top-states",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          <span
                            className="light"
>
                            Colorado:
</span>,
                          " $5,579,829 ",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Colorado:",
                            "className": "light",
                          },
                          "ref": null,
                          "rendered": "Colorado:",
                          "type": "span",
                        },
                        " $5,579,829 ",
                      ],
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          <span
                            className="light"
>
                            California:
</span>,
                          " $345,014 ",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "California:",
                            "className": "light",
                          },
                          "ref": null,
                          "rendered": "California:",
                          "type": "span",
                        },
                        " $345,014 ",
                      ],
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          <span
                            className="light"
>
                            New York:
</span>,
                          " $317,925",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "New York:",
                            "className": "light",
                          },
                          "ref": null,
                          "rendered": "New York:",
                          "type": "span",
                        },
                        " $317,925",
                      ],
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          <span
                            className="light"
>
                            Connecticut:
</span>,
                          " $142,720",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Connecticut:",
                            "className": "light",
                          },
                          "ref": null,
                          "rendered": "Connecticut:",
                          "type": "span",
                        },
                        " $142,720",
                      ],
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          <span
                            className="light"
>
                            Massachussets:
</span>,
                          " $105,922",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Massachussets:",
                            "className": "light",
                          },
                          "ref": null,
                          "rendered": "Massachussets:",
                          "type": "span",
                        },
                        " $105,922",
                      ],
                      "type": "li",
                    },
                  ],
                  "type": "ul",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " *Hover over map to see all totals ",
                    "className": "small",
                  },
                  "ref": null,
                  "rendered": " *Hover over map to see all totals ",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
